### PR TITLE
GG-36614 Duplicated index creation using SQL and already existing QueryFields index leads to node start-up failure

### PR DESCRIPTION
**Pull request by zstan**

### CODE REVIEW

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/opt/GridH2Table.java [hunk 1]
```diff
@@ -1059,48 +1055,51 @@ public boolean rebuildFromHashInProgress() {
...
     /**
      * Add index that is in an intermediate state and is still being built, thus is not used in queries until it is
      * promoted.
      *
      * @param idx Index to add.
+     * @return {@code true} If index not intersects by name with already exists one.
      * @throws IgniteCheckedException If failed.
      */
-    public void proposeUserIndex(Index idx) throws IgniteCheckedException {
+    public boolean proposeUserIndex(Index idx, boolean ifNotExists) throws IgniteCheckedException {
```
Comment (xtern) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/opt/GridH2Table.java:1079]: Where `ifNotExists` flag is used in this method? :thinking:

**Comment by zstan** [comment]
@xtern thanks for review, all issues are fixed, TC in progress. Can i move forward ?

**Comment by xtern** [comment]
Thanks for the contribution, in general it looks like the patch completely solves the main problem described in the issue. 

But as was discussed privately, we need to file a ticket about incorrect `IF NOT EXISTS` behavior, when index name has not been specified.

**Review by xtern — APPROVED** [review]

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/opt/GridH2Table.java [hunk 2]
```diff
@@ -1059,48 +1055,51 @@ public boolean rebuildFromHashInProgress() {
...
+
             Index oldTmpIdx = tmpIdxs.put(idx.getName(), (GridH2IndexBase)idx);
 
             assert oldTmpIdx == null;
         }
         finally {
             unlock(true);
         }
+
+        return true;
```
Comment (AMashenkov) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/opt/GridH2Table.java:1102]: Looks like the method always returns true.

Comment (zstan) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/opt/GridH2Table.java]: got it

**Pull request closed without merging**